table/server/common/src/test/java/alluxio/table/common/udb/UdbAttachSpecTest.java

```diff
@@ -0,0 +1,264 @@
+        .include()
+        .addPartition(TABLE(1), partitionBuilder);
+
+    UdbAttachSpec spec = mBuilder.build();
+    assertTrue(spec.isBypassedTable(TABLE(1)));
+    assertFalse(spec.isFullyBypassedTable(TABLE(1)));
+    assertTrue(spec.isBypassedPartition(TABLE(1), PART(1)));
+    assertTrue(spec.isBypassedPartition(TABLE(1), PART(2)));
+    assertFalse(spec.isBypassedPartition(TABLE(1), PART(3)));
+  }
+
+  @Test
+  public void includedTableNamesOnly() {
+    mBuilder.bypass().include().addName(TABLE(2));
+    UdbAttachSpec spec = mBuilder.build();
+    assertTrue(spec.isBypassedTable(TABLE(2)));
+    assertTrue(spec.isFullyBypassedTable(TABLE(2)));
+    assertTrue(spec.isBypassedPartition(TABLE(2), PART(1)));
+    assertTrue(spec.isBypassedPartition(TABLE(2), PART(2)));
+    assertTrue(spec.isBypassedPartition(TABLE(2), PART(3)));
+  }
+
+  @Test
+  public void includedNonExistentTable() {
+    mBuilder.bypass().include().addName(TABLE(3));
+    UdbAttachSpec spec = mBuilder.build();
+    assertFalse(spec.isBypassedTable(TABLE(4)));
+    assertFalse(spec.isFullyBypassedTable(TABLE(4)));
+    assertFalse(spec.isBypassedPartition(TABLE(4), PART(1)));
+    assertFalse(spec.isBypassedPartition(TABLE(4), PART(2)));
+    assertFalse(spec.isBypassedPartition(TABLE(4), PART(3)));
+  }
+
+  @Test
+  public void includedTablePatterns() {
+    mBuilder.bypass().include().addPattern(Pattern.compile(TABLE("\\d")));
+    UdbAttachSpec spec = mBuilder.build();
+    assertTrue(spec.isBypassedTable(TABLE(1)));
+    assertTrue(spec.isBypassedTable(TABLE(2)));
+    assertTrue(spec.isFullyBypassedTable(TABLE(1)));
+    assertTrue(spec.isFullyBypassedTable(TABLE(2)));
+  }
+
+  @Test
+  public void includedTableMixedNamesPatternsPartitions() {
+    UdbAttachSpec.PartitionSpecBuilder partitionBuilder = new UdbAttachSpec.PartitionSpecBuilder();
+    partitionBuilder
+        .include()
+        .addNames(ImmutableSet.of(PART(1), PART(2)))
+        .addPattern(Pattern.compile(PART("[a-z]")));
+    mBuilder.bypass()
+        .include()
+        .addName(TABLE(1))
+        .addPattern(Pattern.compile(TABLE("[a-z]")))
+        .addPartition(TABLE(2), partitionBuilder);
+
+    UdbAttachSpec spec = mBuilder.build();
+    assertTrue(spec.isBypassedTable(TABLE(1)));
+    assertTrue(spec.isBypassedTable(TABLE(2)));
+    assertTrue(spec.isBypassedTable(TABLE("a")));
+    assertTrue(spec.isFullyBypassedTable(TABLE(1)));
+    assertFalse(spec.isFullyBypassedTable(TABLE(2)));
+    assertTrue(spec.isFullyBypassedTable(TABLE("a")));
+    assertTrue(spec.isBypassedPartition(TABLE(2), PART(1)));
+    assertTrue(spec.isBypassedPartition(TABLE(2), PART(2)));
+    assertTrue(spec.isBypassedPartition(TABLE(2), PART("a")));
+    assertTrue(spec.isBypassedPartition(TABLE("a"), PART(1)));
+    assertTrue(spec.isBypassedPartition(TABLE("a"), PART(2)));
+    assertTrue(spec.isBypassedPartition(TABLE("a"), PART("a")));
+  }
+
+  @Test(expected = IllegalArgumentException.class)
```

Comment:
`assertThrows` is the preferred way nowadays for expecting exception in JUNIT.
https://junit.org/junit5/docs/5.0.1/api/org/junit/jupiter/api/Assertions.html#assertThrows-java.lang.Class-org.junit.jupiter.api.function.Executable-